### PR TITLE
[libc++][doc] Update the release notes for LLVM 19.

### PR DESCRIPTION
**Pull request by mordante**
This is a preparation for the upcoming LLVM 19 release.

### CODE REVIEW

**Comment by mordante** [comment]
Cherry-picking https://github.com/llvm/llvm-project/pull/99061 failed.

**Review by ldionne — APPROVED** [review]

File: libcxx/docs/ReleaseNotes/19.rst [hunk 1]
```diff
@@ -151,10 +175,11 @@ Deprecations and Removals
 - libc++ no longer supports ``std::allocator<const T>`` and containers of ``const``-qualified element type, such
   as ``std::vector<const T>`` and ``std::list<const T>``. This used to be supported as an undocumented extension.
   If you were using ``std::vector<const T>``, replace it with ``std::vector<T>`` instead. The
-  ``_LIBCPP_ENABLE_REMOVED_ALLOCATOR_CONST`` macro can be defined to temporarily re-enable this extension as
-  folks transition their code. This macro will be honored for one released and ignored starting in LLVM 20.
+  ``_LIBCPP_ENABLE_REMOVED_ALLOCATOR_CONST`` macro can be defined to temporarily re-enable this extension.
+  to temporarily re-enable this extension to make it easier to update user code
```
Comment (var-const) [inline libcxx/docs/ReleaseNotes/19.rst:179]: Note: the `temporarily re-enable` part seems to be duplicated.

**Comment by tru** [comment]
Should @var-const comment maybe be fixed before I merge this?

File: libcxx/docs/ReleaseNotes/19.rst (hunk 1 above)
Comment (mordante) [reply libcxx/docs/ReleaseNotes/19.rst]: Thanks nice catch!

**Review by mordante — COMMENTED** [review]
> Should @var-const comment maybe be fixed before I merge this?

Yes. I just pushed a fix.

**Pull request merged**